src/chart/generateCategoricalChart.tsx

```diff
@@ -285,8 +285,19 @@ const getTooltipData = (state: CategoricalChartState, chartData: any[], layout:
   const activeIndex = calculateActiveTickIndex(pos, ticks, tooltipTicks, axis);
 
   if (activeIndex >= 0 && tooltipTicks) {
-    const activeLabel = tooltipTicks[activeIndex] && tooltipTicks[activeIndex].value;
-    const activePayload = getTooltipContent(state, chartData, activeIndex, activeLabel);
+    const activeLabel = tooltipTicks[activeIndex] && tooltipTicks[activeIndex].value; // for backward compatibility
+    let activePayload: any[];
+    if (state.xAxisMap) {
```

Comment:
thinking about this again, this does not work if you have a chart in the opposite layout.

Trying it, this causes a pretty big regression where the tooltip does not show at all

Reply:
Nice catch. I didn't think of the vertical layout. Will add unit tests for both vertical and horizontal layouts.

Reply:
What's interesting is that the regular "Vertical" storybook seems to work fine and show the tooltip as expected. So it's only when there are multiple axis that things go wrong apparently.

I have been trying to write tests for the usecases that you mentioned but I am struggling (first time using react testing library). 

Here is what I was hoping to do:
- Copy the following test: "Renders tooltip when Tooltip item is added"
- Add extra x-axis, y-axis and area to mimic the storybook that you posted
- Fire a mouseover event on one of the area elements: `fireEvent.mouseOver(area)`
- Check for visibility of the tooltip wrapper: `expect(tooltipWrapper).toBeVisible()`

Unfortunately, this doesn't seem to work. I used the testing library screen.debug() to check the state of the dom but there is no change. Also, I added the storybook that you posted and I don't see any mouse event listener on any element so I am a bit confused. :thinking: 

I can't push broken tests so unless I copy the whole thing here you won't see my attempt.

Reply:
You can push broken tests with --no-verify